### ISSUE TITLE
Does halogen support something like the React.Fragment

### ISSUE THREAD

**Issue by shd101wyy** [purescript-halogen/purescript-halogen#712]
Does halogen support something like the React.Fragment? Thank you!

https://reactjs.org/docs/fragments.html

**Comment by thomashoneyman** [comment]
Sorry, @shd101wyy, but Halogen does not support fragments. I would like to add fragments to Halogen when I have some time -- fragments are certainly possible to add to the library, but the implementation will take some care to ensure there's not a performance hit to the library as a whole.

In the meantime most folks use a wrapping `<div>` or other element as a workaround.

**Comment by shd101wyy** [comment]
Yeah it would be awesome if we could support this feature :smile: 
Thanks a lot :+1:

**Comment by garyb** [comment]
This is something that will primarily need doing in https://github.com/purescript-halogen/purescript-halogen-vdom rather than in Halogen itself. Keeping this issue is fine though, since it'll need a `fragment` constructor or something adding here, but most of the work will be in vdom.

**Comment by codingedgar** [comment]
Is there any way to return an array from a component?

**Comment by garyb** [comment]
No, we'd need fragment in vdom for that too I think.

**Comment by garyb** [comment]
I'm afraid not.

**Comment by jquesada2016** [comment]
I know this isn't a true workaround, but I did find _a_ workaround that might be helpful to someone that's needing this, especially when you need to have a component returning a fragment within a flex or grid container.

To emulate a fragment, your component can return a `span` with `display: contents`, and that'll do the trick!
